rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java

```diff
@@ -0,0 +1,534 @@
+import co.rsk.crypto.Keccak256;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static co.rsk.peg.PegTestUtils.createFederation;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class PegUtilsTest {
+    private static final BridgeConstants bridgeMainnetConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcMainnetParams = bridgeMainnetConstants.getBtcParams();
+    private static final Context context = new Context(bridgeMainnetConstants.getBtcParams());
+    ActivationConfig.ForBlock activations = ActivationConfigsForTest.tbd600().forBlock(0);
+
+    BridgeStorageProvider provider;
+
+    private static final int FIRST_OUTPUT_INDEX = 0;
+    private static final int FIRST_INPUT_INDEX = 0;
+
+    @BeforeEach
+    void init() {
+        provider = mock(BridgeStorageProvider.class);
+    }
+
+    @Test()
+    void test_getTransactionType_before_tbd_600() {
+        // Arrange
+        ActivationConfig.ForBlock fingerrootActivations = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        Wallet liveFederationWallet = mock(Wallet.class);
+        BtcTransaction btcTransaction = mock(BtcTransaction.class);
+
+        // Act
+        RegisterBtcTransactionException exception = Assertions.assertThrows(RegisterBtcTransactionException.class, () -> {
+            PegUtils.getTransactionType(fingerrootActivations, provider, liveFederationWallet, btcTransaction);
+        });
+
+        // Assert
+        String expectedMessage = "Can't call PegUtil.getTransactionType before RSKIP379 activation.";
+        Assertions.assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    void test_getTransactionType_tx_sending_funds_to_unknown_address() throws RegisterBtcTransactionException {
+        // Arrange
+        Federation activeFederation = bridgeMainnetConstants.getGenesisFederation();
+        Wallet liveFederationWallet = new BridgeBtcWallet(context, Collections.singletonList(activeFederation));
+
+        Address unknownAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "unknown");
+
+        Coin minimumPeginTxValue = bridgeMainnetConstants.getMinimumPeginTxValue(activations);
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTransaction.addOutput(minimumPeginTxValue.minus(Coin.SATOSHI), unknownAddress);
```

Comment:
Any reason to send below the minimum? I think it kind of creates a situation where you are testing 2 things at once, below the minimum and unknown receiver

Reply:
Since the previous method to identify the tx type also checks if pegin was valid, I wanted to prove this method doesn't care about the amount of the tx.

Reply:
No reason. Fixed.